buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -936,7 +936,9 @@ class BuildPlugin implements Plugin<Project> {
 
                 executable = "${project.runtimeJavaHome}/bin/java"
                 workingDir = project.file("${project.buildDir}/testrun/${test.name}")
-                maxParallelForks = project.gradle.startParameter.maxWorkerCount
+
+                def defaultParallel = project.rootProject.ext.defaultParallel
+                maxParallelForks = defaultParallel == 'auto' ? Runtime.runtime.availableProcessors() : Integer.parseInt(defaultParallel)
```

Comment:
I think we should ditch "auto" completely,  and just set it to what it meant in the randomized runner `Math.max(4, Runtime.runtime.availableProcessors())` note that setting this  to max processors is know to cause test failures. Many tests start ES nodes and becomes too much.

Reply:
I have ditched "auto" if you look at the latest commits. I've updated this to cap parallel forks at 4 if we can't determine the number of physical cpus.

Reply:
Actually, I've gone with `availableProcessors() / 2` as I think that's going to be pretty safe in most cases.